compiler/rustc_middle/src/hir/map.rs

```diff
@@ -1044,23 +1042,23 @@ impl<'hir> Map<'hir> {
         }
     }
 
-    pub fn span_if_local(self, id: DefId) -> Option<Span> {
-        id.is_local().then(|| self.tcx.def_span(id))
+    pub fn hir_span_if_local(self, id: DefId) -> Option<Span> {
+        id.is_local().then(|| self.def_span(id))
     }
```

Comment:
followup once this PR lands

I think we should remove this method. It's just a wrapper for `def_span` which doesn't return stuff if the definition isn't local. I feel like this behavior should be explicit and that most places actually should to use the span regardless of whether it's local :thinking: